compiler/rustc_infer/src/infer/outlives/obligations.rs

```diff
@@ -80,19 +80,47 @@ use crate::infer::{self, GenericKind, InferCtxt, RegionObligation, SubregionOrig
 use crate::traits::{ObligationCause, ObligationCauseCode};
 
 impl<'tcx> InferCtxt<'tcx> {
+    pub fn outlives_predicate_with_cause(
```

Comment:
I chose to name it `InferCtxt::outlives_predicate_with_cause` to match the naming of `InferCtxt::subtype_predicate`, etc. Notably, there's no verb in these. I could add one I guess (`register_*`) but I didn't think it was really necessary.